base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1066 @@
+    
+    if (bv.RecycleClean == TRUE)
+    {
+        FileStruct.pFrom = NULL;
+        FileStruct.pFrom = wcv.RecycleBinDir;
+
+        SHFileOperationW(&FileStruct);
+        
+        SendMessageW(HProgressBar, PBM_SETPOS, 50, 0);
+        SetDlgItemTextW(dv.hwndDlg, IDC_STATIC_INFO, L"Recycled files");
+        Sleep(150);
+    }
+    
+    if (bv.ChkDskClean == TRUE)
+    {
+        for(int i = 0; i < 10; i++)
+        {
+            StringCchPrintfW(TargetedDir, _countof(TargetedDir), L"%s\\FOUND.%.3i", wcv.DriveLetter, i);
+            if (PathIsDirectoryW(TargetedDir))
+            {
+                FileStruct.pFrom = NULL;
+                FileStruct.pFrom = TargetedDir;
+
+                SHFileOperationW(&FileStruct);
+                Sleep(150);
+            }
+        
+            else
+                break;
+        }
+        SendMessageW(HProgressBar, PBM_SETPOS, 75, 0);
+        SetDlgItemTextW(dv.hwndDlg, IDC_STATIC_INFO, L"Old ChkDsk files");
+    }
+
+    if (bv.RappsClean == TRUE)
+    {
+        FileStruct.pFrom = NULL;
+        FileStruct.pFrom = wcv.RappsDir;
+
+        SHFileOperationW(&FileStruct);
+        SendMessageW(HProgressBar, PBM_SETPOS, 100, 0);
+        SetDlgItemTextW(dv.hwndDlg, IDC_STATIC_INFO, L"Temporary RAPPS Files");
+    }
+    
+    SendMessageW(dv.hwndDlg, WM_DESTROY, 0, 0);
+
+    return TRUE;
+}
+
+uint64_t DirSizeFunc(PWCHAR TargetDir)
+{
+    WIN32_FIND_DATAW DataStruct;
+    ZeroMemory(&DataStruct, sizeof(DataStruct));
+    LARGE_INTEGER sz;
+    ZeroMemory(&sz, sizeof(sz));
+    HANDLE HandleDir = NULL;
+    uint64_t Size = 0;
+    WCHAR TargetedDir[2048] = { 0 };
+    WCHAR ReTargetDir[2048] = { 0 };
+    DWORD DwError = 0;
+    
+    StringCchPrintfW(TargetedDir, _countof(TargetedDir), L"%s%s", TargetDir, L"\\*");
+    HandleDir = FindFirstFileW(TargetedDir, &DataStruct);
+    DwError = GetLastError();
+    
+    if (DwError != ERROR_NO_MORE_FILES && HandleDir == INVALID_HANDLE_VALUE)
+    {
+        if (DwError == ERROR_PATH_NOT_FOUND)
+        {
+            return 0;
+        }
+
+        else
+        {
+            MessageBoxW(NULL, L"FindFirstFileW() failed!", L"Error", MB_OK | MB_ICONERROR);
+            return 1;
+        }
+    }
+
+    do
+    {
+        if (wcscmp(DataStruct.cFileName, L".") != 0 && wcscmp(DataStruct.cFileName, L"..") != 0)
+        {
+            if (DataStruct.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
+            {
+                StringCchPrintfW(ReTargetDir, _countof(ReTargetDir), L"%s\\%s", TargetDir, DataStruct.cFileName);
+                Size = DirSizeFunc(ReTargetDir);
+            }
+
+            else
+            {
+                sz.LowPart = DataStruct.nFileSizeLow;
+                sz.HighPart = DataStruct.nFileSizeHigh;
+                Size += sz.QuadPart;
+            }
+        }
+    }
+
+    while (FindNextFileW(HandleDir, &DataStruct));
+
+    FindClose(HandleDir);
+
+    return Size;
+}
+
+BOOL FindRecycleBin(PWCHAR TargetDrive)
```

Comment:
I am not sure where to even start here.....

What is wrong with using `CSIDL_BITBUCKET` to get the recycle bin?
or if you want to delete from the recycle bin, use `SHEmptyRecycleBinW` ?

Reply:
and `SHQueryRecycleBinW` if you want to know what's in it

Reply:
@learn-more Isn't `SHQueryRecycleBinW` a stub in ReactOS?

Reply:
it might be a stub, but I'd say it is far better than this construction.
So the best approach would be to make it work in windows with this function, and then we can see about implementing this missing function in reactos.
most code already seems to be there so shouldnt take more than an hour or so.